src/network/networkd-dhcp4.c

```diff
@@ -1795,6 +1801,44 @@ int config_parse_dhcp_mud_url(
         return free_and_strdup_warn(&network->dhcp_mudurl, unescaped);
 }
 
+int config_parse_dhcp_lease_lifetime(const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network = userdata;
+        unsigned k;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                network->dhcp_lease_lifetime = 0;
+                return 0;
+        }
+
+        /* We accept only "forever" or "infinity". */
+        if (STR_IN_SET(rvalue, "forever", "infinity"))
+                k = CACHE_INFO_INFINITY_LIFE_TIME;
```

Comment:
if only two values are valid here, i think we should store this as a bool, to make this clear.